shell/common/shell_unittests.cc

```diff
@@ -600,5 +600,82 @@ TEST_F(ShellTest, WaitForFirstFrameInlined) {
   ASSERT_FALSE(event.WaitWithTimeout(fml::TimeDelta::FromMilliseconds(1000)));
 }
 
+TEST_F(ShellTest, SetResourceCacheSize) {
+  Settings settings = CreateSettingsForFixture();
+  auto task_runner = GetThreadTaskRunner();
+  TaskRunners task_runners("test", task_runner, task_runner, task_runner,
+                           task_runner);
+  std::unique_ptr<Shell> shell =
+      CreateShell(std::move(settings), std::move(task_runners));
+
+  // Create the surface needed by rasterizer
+  PlatformViewNotifyCreated(shell.get());
+
+  auto configuration = RunConfiguration::InferFromSettings(settings);
+  configuration.SetEntrypoint("emptyMain");
+
+  RunEngine(shell.get(), std::move(configuration));
+  PumpOneFrame(shell.get());
+
+  EXPECT_EQ(shell->GetRasterizer()->GetResourceCacheMaxBytes(),
+            static_cast<size_t>(24 * (1 << 20)));
+
+  fml::TaskRunner::RunNowOrPostTask(
+      shell->GetTaskRunners().GetPlatformTaskRunner(), [&shell]() {
+        shell->GetPlatformView()->SetViewportMetrics(
+            {1.0, 400, 200, 0, 0, 0, 0, 0, 0, 0, 0});
+      });
+  PumpOneFrame(shell.get());
+
+  EXPECT_EQ(shell->GetRasterizer()->GetResourceCacheMaxBytes(), 3840000U);
+
+  std::string request_json =
+      "{\"method\": \"Skia.setResourceCacheMaxBytes\", \"args\": 10000}";
+  std::vector<uint8_t> data(request_json.begin(), request_json.end());
+  auto platform_message = fml::MakeRefCounted<PlatformMessage>(
+      "flutter/skia", std::move(data), nullptr);
+  SendEnginePlatformMessage(shell.get(), std::move(platform_message));
```

Comment:
Can you make sure a reply is received for this message? I think that's broken right now.

Reply:
There's no reply on this mesage.  It's not implemented at all.  What's the use case for it?

Reply:
See b/137134910 for details. Without a reply applications can't await setting this, so have no visibility into when the change has taken effect. I think sending an empty message around here would fix it: https://github.com/flutter/engine/blob/d937b693f556db17cf80622e6fdffb7e27316f61/shell/common/shell.cc#L945

`message->response->CompleteEmpty()` should be enough.

Reply:
I've added an empty reply and a test for it.

Reply:
in https://github.com/flutter/engine/pull/10434/commits/b50cab1ba9f4fa4686798007898e89c9383f06d0